api/app/Http/Requests/Auth/RegisterRequest.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\Http\Requests\Auth;
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class RegisterRequest extends FormRequest
+{
+    public function authorize(): bool
+    {
+        return true;
+    }
+
+    public function rules(): array
+    {
+        return [
+            'name' => ['required', 'string'],
+            'email' => ['required', 'string', 'unique:App\Entities\User,email'],
+            'password' => ['required', 'string'],
+            'nickname' => ['sometimes', 'string', 'nullable', 'unique:App\Entities\User,nickname']
+        ];
+    }
```

Comment:
I'd recommend using the `Rule` class helper:

```suggestion
        return [
            'name' => ['required', 'string'],
            'email' => [
                'required',
                'string',
                Rule::unique(User::class,'email'),
            ],
            'password' => ['required', 'string'],
            'nickname' => [
                'sometimes',
                'string',
                'nullable',
                Rule::unique(User::class,'nickname'),
            ],
        ];
```